### PR TITLE
Add benchmark for ShaderMask RasterCache

### PR DESCRIPTION
**Pull request by wangying3426**
Benchmark for https://github.com/flutter/engine/pull/32027
A set of test data on Redmi Note 3 in profile mode is：

 1. Benchmark A/B Test

```dart
═════════════════════════╡ ••• Final A/B results ••• ╞══════════════════════════

Score	Average A (noise)	Average B (noise)	Speed-up
average_frame_build_time_millis	2.25 (0.00%)	2.20 (0.00%)	1.02x	
worst_frame_build_time_millis	4.23 (0.00%)	3.93 (0.00%)	1.08x	
90th_percentile_frame_build_time_millis	3.02 (0.00%)	2.95 (0.00%)	1.02x	
99th_percentile_frame_build_time_millis	4.05 (0.00%)	3.15 (0.00%)	1.29x	
average_frame_rasterizer_time_millis	6.83 (0.00%)	3.22 (0.00%)	2.12x	
worst_frame_rasterizer_time_millis	8.63 (0.00%)	5.22 (0.00%)	1.65x	
90th_percentile_frame_rasterizer_time_millis	7.94 (0.00%)	3.72 (0.00%)	2.13x	
99th_percentile_frame_rasterizer_time_millis	8.38 (0.00%)	4.46 (0.00%)	1.88x	
average_layer_cache_count	0.00 (0.00%)	2.00 (0.00%)	0.00x	
90th_percentile_layer_cache_count	0.00 (0.00%)	2.00 (0.00%)	0.00x	
99th_percentile_layer_cache_count	0.00 (0.00%)	2.00 (0.00%)	0.00x	
worst_layer_cache_count	0.00 (0.00%)	2.00 (0.00%)	0.00x	
average_layer_cache_memory	0.00 (0.00%)	3.68 (0.00%)	0.00x	
90th_percentile_layer_cache_memory	0.00 (0.00%)	3.68 (0.00%)	0.00x	
99th_percentile_layer_cache_memory	0.00 (0.00%)	3.68 (0.00%)	0.00x	
worst_layer_cache_memory	0.00 (0.00%)	3.68 (0.00%)	0.00x	
average_picture_cache_count	2.00 (0.00%)	0.00 (0.00%)	Infinityx	
90th_percentile_picture_cache_count	2.00 (0.00%)	0.00 (0.00%)	Infinityx	
99th_percentile_picture_cache_count	2.00 (0.00%)	0.00 (0.00%)	Infinityx	
worst_picture_cache_count	2.00 (0.00%)	0.00 (0.00%)	Infinityx	
average_picture_cache_memory	3.67 (0.00%)	0.00 (0.00%)	Infinityx	
90th_percentile_picture_cache_memory	3.67 (0.00%)	0.00 (0.00%)	Infinityx	
99th_percentile_picture_cache_memory	3.67 (0.00%)	0.00 (0.00%)	Infinityx	
worst_picture_cache_memory	3.67 (0.00%)	0.00 (0.00%)	Infinityx	
new_gen_gc_count	0.00 (0.00%)	0.00 (0.00%)	NaNx	
old_gen_gc_count	0.00 (0.00%)	0.00 (0.00%)	NaNx
```

2. Memory

- NotCached
Screen Shot 2022-03-15 at 10 56 46 PM

- Cached
Screen Shot 2022-03-15 at 10 52 20 PM

Associated with https://github.com/flutter/flutter/pull/99542

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by flar — APPROVED** [review]
LGTM

Running on my Moto G4 the layout follows the same pattern as the other cache perf benchmark.

**Comment by flar** [comment]
It looks like the PR needs to be rebased to solve some merge conflicts and you also seem to have an issue with the order of the imports in main.dart?

**Comment by wangying3426** [comment]
> It looks like the PR needs to be rebased to solve some merge conflicts and you also seem to have an issue with the order of the imports in main.dart?

Yes, I have resolved the conflicts.

File: .ci.yaml [hunk 1]
```diff
@@ -1527,6 +1527,18 @@ targets:
...
+  - name: Linux_android shader_mask_cache_perf__e2e_summary
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      tags: >
+        ["devicelab","android","linux"]
+      task_name: shader_mask_cache_perf__e2e_summary
+      benchmark: "true"
```
Comment (CaseyHillers) [inline .ci.yaml:1539]: nit: You can rm this as it's no longer used

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**